test/IPSpec.hs

```diff
@@ -42,6 +45,18 @@ arbitraryIIPv6Str adrGen msklen = toIv6 <$> adrGen <*> lenGen
     toIv6 adr len = Iv6 $ show adr ++ "/" ++ show len
     lenGen = oneof [choose (minBound, -1), choose (msklen + 1, maxBound)]
 
+data PaddedIPv4 = PaddedIPv4 [Int] [Int]
+
+instance Show PaddedIPv4 where
+  show (PaddedIPv4 pads digs) = intercalate "." $ zipWith (\p d -> replicate p '0' <> show d) pads digs
+
+instance Arbitrary PaddedIPv4 where
+  arbitrary = PaddedIPv4 <$> replicateM 4 (choose (0, 4)) <*> replicateM 4 (choose (0, 255))
+  shrink (PaddedIPv4 pads digs) = (PaddedIPv4 <$> shrinkQuad pads <*> pure digs) <> (PaddedIPv4 pads <$> shrinkQuad digs)
```

Comment:
When shrinking both the prefix pads and the octet digits, you end up with failure like `0.00.0.0` which isn't awful.  If I don't shrink the digits, you end up with failures like `192.102.137.070` which look prettier, but the shrinking seems like a good idea overall.